pkg/operator/etcdmemberipmigrator/etcdmemberipmigrator.go

```diff
@@ -52,33 +44,26 @@ func NewEtcdMemberIPMigrator(
 	networkInformer configv1informers.NetworkInformer,
 	etcdClient etcdcli.EtcdClient,
 	eventRecorder events.Recorder,
-) *EtcdMemberIPMigrator {
+) factory.Controller {
 	c := &EtcdMemberIPMigrator{
 		operatorClient:       operatorClient,
 		etcdClient:           etcdClient,
 		nodeLister:           kubeInformers.Core().V1().Nodes().Lister(),
 		infrastructureLister: infrastructureInformer.Lister(),
 		networkLister:        networkInformer.Lister(),
-
-		cachesToSync: []cache.InformerSynced{
-			operatorClient.Informer().HasSynced,
-			kubeInformers.Core().V1().Nodes().Informer().HasSynced,
-			infrastructureInformer.Informer().HasSynced,
-			networkInformer.Informer().HasSynced,
-			operatorClient.Informer().HasSynced,
-		},
-		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "EtcdMemberIPMigrator"),
-		eventRecorder: eventRecorder.WithComponentSuffix("etcd-member-ip-migrator"),
 	}
-	kubeInformers.Core().V1().Nodes().Informer().AddEventHandler(c.eventHandler())
-	networkInformer.Informer().AddEventHandler(c.eventHandler())
-	operatorClient.Informer().AddEventHandler(c.eventHandler())
 
-	return c
+	return factory.New().ResyncEvery(time.Minute).WithInformers(
+		operatorClient.Informer(),
+		kubeInformers.Core().V1().Nodes().Informer(),
+		infrastructureInformer.Informer(),
```

Comment:
added infrastructure informer I guess we should of had that before.

Reply:
yeah, I think that was a bug :-)